### PR TITLE
Implement per-user daily usage limits

### PR DESCRIPTION
**Pull request by vasiliadi**
## Summary

- Each user now has an individual `daily_limit` stored in the DB (default `0` = blocked until admin grants access, same workflow as flipping `approved`).
- Redis tracks per-user counters under `RPD:{user_id}`; the nightly cron resets them with SCAN + DEL on `RPD:*`.
- RPM (per-minute) throttle remains **global** and unchanged.

## What changed

| File | Change |
|------|--------|
| `src/models.py` + migration | Add `daily_limit` column (`server_default=0`) |
| `src/config.py` | Drop `per_day_limit` / `DAILY_LIMIT=20`; expose shared `rate_limiter_store` |
| `src/services.py` | Rewrite `check_quota(user_id, daily_limit, quantity)` — no DB roundtrip; add `get_remaining_quota` |
| `src/summary.py` | Thread `user_id` and `daily_limit` through all four `summarize_*` functions |
| `src/handlers.py` | Pass `user_id=user.user_id, daily_limit=user.daily_limit` at every call site |
| `src/main.py` | Add `/limit` command; show `daily_limit` + `remaining quota` in `/myinfo` |
| `scripts/cron.py` | Rewrite to plain `redis` SCAN + DEL on `RPD:*`; drop `rush[redis]` dependency |
| `tests/` | Update all existing tests; add quota-isolation tests in `test_services.py` |

## Reviewer notes

- Admin sets `daily_limit` directly in the DB — no bot command needed (mirrors the `approved` workflow).
- Mid-day cap changes take effect on the next request without resetting the Redis counter.
- New users start with `daily_limit=0` and are blocked until the admin sets a positive value.
- All pre-commit hooks (ruff, pytest) pass.

## Test plan

- [x] Apply migration: `alembic upgrade head` — confirm `daily_limit` column with default `0`
- [x] Register fresh user → trigger summarization → expect daily limit exceeded
- [x] Set `daily_limit=2` in DB → send 3 summaries → first two succeed, third blocked
- [x] Check Redis: `redis-cli KEYS 'RPD:*'` shows per-user keys; `RPM` remains a single global key
- [x] Run cron locally → all `RPD:*` keys deleted; next request succeeds up to cap
- [x] `uv run pytest` passes

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **New Features**
  * Added `/limit` and enhanced `/myinfo` to show per-user daily limit and remaining quota; per-user daily limits enforced.

* **Chores**
  * Database migration to store per-user daily limits; rate-limiting refactored to per-user keys and direct Redis handling.

* **Documentation**
  * README updated with new bot commands and removal of outdated “Known problems” section.

* **Tests**
  * Expanded coverage for per-user quotas, command behaviors, quota edge cases, and cleanup/error paths.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -71,6 +71,8 @@ set_prompt_strategy - Choose which prompt strategy to use for summary
 toggle_transcription - Toggle transcription for summary (fallback on failure)
 toggle_yt_transcription - Toggle YouTube transcription
 set_target_language - Choose which language you want to translate into
+limit - Show remaining limit
+myinfo - Show my settings
```
[the remark this answers is not in the document]
Comment (vasiliadi) [reply README.md:75]: This is fine.

File: migrations/versions/37d47c20352c_add_daily_limit_to_users.py [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+
+def upgrade() -> None:
+    # ### commands auto generated by Alembic - please adjust! ###
+    op.add_column(
+        "users",
+        sa.Column(
+            "daily_limit", sa.Integer(), server_default="0", nullable=False
+        ),
+    )
```
[the remark this answers is not in the document]
Comment (vasiliadi) [reply migrations/versions/37d47c20352c_add_daily_limit_to_users.py:28]: This is fine.

**Comment by vasiliadi** [comment]
@coderabbitai review

File: src/summary.py [hunk 3]
```diff
@@ -91,27 +96,29 @@ def summarize_with_file(
     if audio_file.name is None or audio_file.uri is None:
         raise AttributeError
     audio_file_name = audio_file.name
```
[the remark this answers is not in the document]
Comment (vasiliadi) [reply src/summary.py]: Thanks, done.

**Comment by vasiliadi** [comment]
@coderabbitai review

File: src/summary.py [hunk 4]
```diff
@@ -81,37 +85,41 @@ def summarize_with_file(
...
+                ),
+            ],
+            config=get_gemini_config(target_language, model=model),
+        )
+        if response.text is None:
+            raise AttributeError
+        return response.text
+    finally:
+        if audio_file_name is not None:
+            gemini_client.files.delete(name=audio_file_name)
```
[the remark this answers is not in the document]
Comment (vasiliadi) [reply src/summary.py]: Done.

File: scripts/cron.py [hunk 5]
```diff
@@ -1,47 +1,35 @@
...
+    rate_limiter_url = f"{os.environ['REDIS_URL']}/0"
+    client = redis_lib.StrictRedis.from_url(url=rate_limiter_url, decode_responses=True)
+
+    batch: list[str] = []
+    deleted = 0
+    for key in client.scan_iter(match=f"{daily_limit_key}:*", count=500):
+        batch.append(key)
+        if len(batch) >= 500:
+            client.unlink(*batch)
+            deleted += len(batch)
```
[the remark this answers is not in the document]
Comment (vasiliadi) [reply scripts/cron.py]: Make sense.

**Comment by vasiliadi** [comment]
@kilo-code-bot review
